### PR TITLE
#12785 prevent NPE on 2 shopware locations pointing to same metasfresh C_BPartner_Location_ID
Prevent import error when company name is unset in Shopware

### PR DESCRIPTION
**Pull request by adi-stefan**
https://github.com/metasfresh/metasfresh/issues/12785

### CODE REVIEW

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v1/bpartner/bpartnercomposite/jsonpersister/JsonPersisterService.java [hunk 1]
```diff
@@ -638,10 +638,12 @@ else if (isUpdateRemove)
 		if (jsonBPartner.isCompanyNameSet())
 		{
 			bpartner.setCompanyName(StringUtils.trim(jsonBPartner.getCompanyName()));
+			bpartner.setCompany(!Check.isBlank(jsonBPartner.getCompanyName()));
```
Comment (metas-ts) [inline backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v1/bpartner/bpartnercomposite/jsonpersister/JsonPersisterService.java:641]:
```suggestion
			bpartner.setCompany(Check.isNotBlank(jsonBPartner.getCompanyName()));
```

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/jsonpersister/JsonPersisterService.java [hunk 2]
```diff
@@ -797,6 +797,7 @@ private BooleanWithReason syncJsonToBPartner(
 		if (jsonBPartner.isCompanyNameSet())
 		{
 			bpartner.setCompanyName(StringUtils.trim(jsonBPartner.getCompanyName()));
+			bpartner.setCompany(!Check.isBlank(jsonBPartner.getCompanyName()));
```
Comment (metas-ts) [inline backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/bpartner/bpartnercomposite/jsonpersister/JsonPersisterService.java:800]:
```suggestion
			bpartner.setCompany(Check.isNotBlank(jsonBPartner.getCompanyName()));
```

**Review by metas-ts — COMMENTED** [review]
please check my small suggestions

**Review by metas-ts — APPROVED** [review]

File: backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/order/sales/OrderService.java [hunk 3]
```diff
@@ -83,6 +85,11 @@ public void createOrderPayment(@NonNull final JsonOrderPaymentCreateRequest requ
 		{
 			throw new AdempiereException("Cannot find the orgId from either orgCode=" + request.getOrgCode() + " or the current user's context.");
 		}
+		final ExternalId paymentExternalId = ExternalId.ofOrNull(request.getExternalPaymentId());
+		if (paymentExternalId != null && paymentDAO.getByExternalId(paymentExternalId, orgId).isPresent())
+		{
+			throw new AdempiereException("Payment with externalId=" + paymentExternalId + " already exists in orgId=" + orgId);
```
[the remark this answers is not in the document]
Comment (adi-stefan) [reply backend/de.metas.business.rest-api-impl/src/main/java/de/metas/rest_api/v2/order/sales/OrderService.java]: addressed to silently fail the insert.

**Review by metas-ts — APPROVED** [review]

**Pull request merged**